enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,295 @@
+ │ dhcpv4_options: 1234 │            │ dhcpv4_options: 1234 │
+ │ address:             │            │ address:             │    
+ │  0a:58:0a:f4:00:01   │            │  0a:58:0a:f4:00:01   │
+ │  10.244.0.8          │            │  10.244.0.8          │
+ └──────────────────────┘            └──────────────────────┘
+
+┌─────────────────────────────────┐
+│ dhcp-options 1234               │
+│   lease_time: 3500              │
+│   router: 169.254.1.1           │
+│   dns_server: [kubedns]         │
+│   server_id: 169.254.1.1        │
+│   server_mac: c0:ff:ee:00:00:01 │
+└─────────────────────────────────┘
+```
+
+**Virt-launcher pod address:**
+ The CNI will not set an address at virt-launcher pod netns, that address is
+ assigned to the VM with the DHCP options from the LSP, this allows to use 
+ kubevirt bridge binding with pod networking and still do live migration.
+
+#### IPAM
+
+The /32 routing feature allows an address to be running at a node different from
+the one "owning" the subnet the address is coming from. This will happen
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what ip pool the VM belongs 
+(using an annotation or searching for subnets) and reserve the address.
+
+#### Detecting hypershift VMs and live migration
+
+This enhancement only applies for hypershift kubevirt workers so the rest of 
+virtual machine should not use it, those VMs will be marked with label 
+`kubevirt.io/allow-pod-network-bridge-binding-live-migration` by hypershift
+kubevirt provider and ovn-k will do /32 routing if detected, the of non
+hypershift VM should not add that special label.
+
+There is also a moment where routing need to do the "switch", meaning that the
+vm has to be routed over the host is running on, to detect that the label
+"kubevirt.io/nodeName" can be check since it appears just after live migration
+has finish. Also to discard old virt-launcher pod the creation timestamp can
+be used, ignoring them is they are not the latest one.
+
+### Risks and Mitigations
+
+### OVN extended arp_proxy features ####
+
+
+The current ovn arp_proxy is missing the following features: enforcing the mac 
+address and ipv6, ideally those get implemented and ovn-kubernetes have 
+them for 4.14 release.
+
+In case they don't arrive on time, it will not be able to configure IPv6 hosted
+clusters at hypershift and VMs arp table will need to be refreshed after live
+migration. 
+
+### Open Questions 
```

Comment:
Maybe it's just me, but I don't follow these open questions: all of them have perfectly valid answers on them. 

I could have misinterpreted the goal of this section though - do not act upon my comment yet please.

Reply:
ditto